content/source/docs/extend/resources.html.md

```diff
@@ -0,0 +1,368 @@
+
+```go
+package example
+
+import (
+    "fmt"
+
+    "github.com/hashicorp/terraform/helper/resource"
+    "github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceExampleInstance() *schema.Resource {
+    return &schema.Resource{
+        Create: resourceExampleInstanceCreate,
+        Read:   resourceExampleInstanceRead,
+        Update: resourceExampleInstanceUpdate,
+        Delete: resourceExampleInstanceDelete,
+
+        Schema: map[string]*schema.Schema{
+            "name": {
+                Type:     schema.TypeString,
+                Required: true,
+            },
+        },
+        Timeouts: &schema.ResourceTimeout{
+            Create: schema.DefaultTimeout(45 * time.Minute),
+        },
+    }
+}
+```
+
+In the above example we see the usage of the timeouts in the schema being configured for what is deemed the appropriate amount of time for the `Create` function. `Read`, `Update`, and `Delete` are also configurable as well as a `Default`. These configured timeouts can be fetched later in the CRUD functions from the passed in `*schema.ResourceData`.
+
+### Retry
+
+A common case for requiring retries or polling is when the backend infrastructure being provisioned is designed to be asynchronous, requiring the developer to repeatedly check the status of the resource. The retry helper takes a timeout and a function that is retried repeatedly. The timeout can be retrieved from the `*schema.ResourceData` struct, using the `Timeout` method, passing in the appropriate timeout key (`scheme.TimeoutCreate`). The retry function provided should return either a `resource.NonRetryableError` for unexpected errors or states, otherwise continue to retry with a `resource.RetryableError`. In the context of a `CREATE` function, once the backend responds with the desired state, finish the function with a `resource.NonRetryableError` wrapping the `READ` function (anything that goes wrong in there is considered unexpected).
+
+```go
+func resourceExampleInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+    name := d.Get("name").(string)
+    client := meta.(*ExampleClient)
+    resp, err := client.CreateInstance(name)
+
+    if err != nil {
+        return fmt.Errorf("Error creating instance: %s", err)
+    }
+
+    return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+        resp, err := client.DescribeInstance(name)
+
+        if err != nil {
+            return resource.NonRetryableError(fmt.Errorf("Error describing instance: %s", err))
+        }
+
+        if resp.Status != "CREATED" {
+            return resource.RetryableError(fmt.Errorf("Expected instance to be created but was in state %s", resp.Status))
+        }
+
+        return resource.NonRetryableError(resourceExampleInstanceRead(d, meta))
+    })
+}
+```
+
+### StateChangeConf
```

Comment:
I think the header should be something like `Retry Helpers`?

Reply:
The higher-level one, sorry for the confusion. I think headers should use a human-friendly language - something the reader can understand and then we can dive into terms like `StateChangeConf` below.